Split marriage-abroad outcome for opposite sex marriages in CNI countries when residing in UK or ceremony country

chrisroos opened the pull request:
This branch splits the outcome_opposite_sex_marriage_in_consular_cni_countries_when_residing_in_uk_or_ceremony_country.govspeak.erb template into two:

* One for residents of the UK
* One for residents of the ceremony country

Moving logic out of the templates and into the flow should leave us with templates that are easier to maintain.

The change in this branch is a precursor to further splitting these two templates by country.

## Expected changes

None. This is a refactoring and doesn't change the behaviour of the system.

floehopper commented:
Other than the fact that this branch has conflicts which need resolving, this looks good to me. Great work! 👍

chrisroos commented:
I've rebased this branch on master and force pushed in preparation for merging.

I've split the two commits that were removing unnecessary code from the two new outcome templates. I had to redo this work as part of resolving the rebasing conflicts and felt that splitting the commits made it slightly easier to follow (thanks for the suggestion, @floehopper). I used the regression tests for _all_ countries to give me confidence that splitting these commits didn't break anything.
